File: Segmentor.m
classdef Segmentor < handle
    % Extracts proper window times for a given science segment
    % Grant David Meadors
    % 02012-02-29
    
    properties (SetAccess = private)
        tA
        tB
        tIsPreceded
        tIsFollowed
    end
    
    methods
        function tSegment = Segmentor(T, ii)
            % find the subdivision times for the specific segment
            
            % by looking at an 'ii'th element of 't.startlist'
            tSegment.tA = T.list{1}(ii);
            tSegment.tB = T.list{2}(ii);
            tSegment.tIsPreceded = T.list{3}(ii);
            tSegment.tIsFollowed = T.list{4}(ii);
            
            disp(strcat('Beginning science segment ... ', num2str(ii)))
            % apply the subdivider to that specific segment,
            % which gives thirty-minute subsections out to tsub.
        end
    end
    
end

File: Segmentor.py
class Segmentor:
    # Extracts proper window times for a given science segment
    # Grant David Meadors
    # 02012-07-29 (JD 2456139)
    # g m e a d o r s @ u m i c h . e d u

    def __init__(self, T, ii):
        # Find the subdivision times for the specific segment
        # by looking at an 'ii'th element of 'T.list'
        self.tA = T.list[0][ii]
        self.tB = T.list[1][ii]
        self.tIsPreceded = T.list[2][ii]
        self.tIsFollowed = T.list[3][ii]

        # Apply the subdivider to that specific segment,
        # Which gives 1024 second windows out to tSub.
        print 'Beginning science segment ... ' + str(ii+1)
